bindings/python/install_data/libtorrent/__init__.pyi

```diff
@@ -0,0 +1,1954 @@
+import datetime
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Iterable
+from typing import Iterator
+from typing import List
+from typing import Mapping
+from typing import Optional
+from typing import overload
+from typing import Tuple
+from typing import Union
+
+from typing_extensions import TypedDict
+
+# I created this file by starting with stubgen and modifying its output.
+
+# I recommend that maintainers occasionally re-run stubgen, and compare it to
+# this file. This diff will be too verbose to use for small changes, but it's the
+# only way to ensure coverage.
+
+# To help with the above process, please keep everything in sorted order, to
+# match stubgen's output.
+
+# Typing notes:
+# - This project has a very particular approach to typed enum values. Pay
+#   special attention to which are typed, and which are just ints.
```

Comment:
I (think I)  made all enums convert to plain `int` on the python side. Mostly for simplicity, and because type checking isn't very strong in python anyway. I'd be open to better ways of doing it. Same thing with flag types.

Reply:
Where the enum values are *declared* in python, I found there's an even split between strongly-typed enums (members of a class descending from `int`, like `operation_t`) and weakly-typed (just `int`s, like `torrent_flags`). I think this is the difference between being declared with boost.python's `enum` versus not.

Where enum values are *accepted*, I think they're almost always weakly-typed (accepts any `int`). I found one exception in `torrent_info.add_tracker()` which requires a strongly-typed `tracker_source` enum value and won't accept an `int`.

IMO, accepting weakly-typed `int` everywhere is a good thing.

I also think we could convert `torrent_flags` and others to boost.python `enum`s. I don't think it affects any usage and it would help users write strongly-typed code of their own